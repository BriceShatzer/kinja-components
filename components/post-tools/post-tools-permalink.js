// @flow

import * as React from 'react';
import styled from 'styled-components';

import { DropdownItem } from '../dropdown';
import { EnsureDefaultTheme } from '../theme';
import type { BlogId } from 'kinja-magma/models/Id';
import type { PostToolsCallbacks} from './';
import Pencil from '../icon19/Pencil';
import Embiggen from '../icon19/Embiggen';
import Mail from '../icon19/Mail';
import ExternalLink from '../icon19/ExternalLink';
import Video from '../icon19/Video';
import Bubble from '../icon19/Bubble';
import StarCircle from '../icon19/StarCircle';
import MoveVertical from '../icon19/MoveVertical';

/*
 * Permalink page PostTools dropdown contents
 */
export const PostToolsPermalink = ({
	editClickHandler,
	embiggenClickHandler,
	sendToEditorsClickHandler,
	promoteClickHandler,
	shareClickHandler,
	initializeInArticleAdTools,
	changeInArticleVideoClickHandler,
	toggleConversationToolsClickHandler,
	sharedToBlogId,
	isEmbiggened
}: PostToolsCallbacks & {
	sharedToBlogId: ?BlogId,
	isEmbiggened: boolean
}) => (
	<EnsureDefaultTheme>
		<React.Fragment>
			{editClickHandler && <DropdownItem
				onClick={editClickHandler}
				icon={<Pencil/>}
				title="Edit"
			/>}
			{embiggenClickHandler && <DropdownItem
				isActive={isEmbiggened}
				onClick={() => embiggenClickHandler(isEmbiggened)}
				icon={<Embiggen/>}
				title="Embiggen"
			/>}
			{sendToEditorsClickHandler && <DropdownItem
				onClick={sendToEditorsClickHandler}
				icon={<Mail/>}
				title="Send to Editors"
			/>}
			{promoteClickHandler && <DropdownItem
				onClick={promoteClickHandler}
				icon={<StarCircle/>}
				title="Promote"
			/>}
			{shareClickHandler && <DropdownItem
				onClick={shareClickHandler}
				icon={<ExternalLink/>}
				title={`${sharedToBlogId ? 'Unshare' : 'Share to Kinja'}`}
			/>}
			{initializeInArticleAdTools && <DropdownItem
				onClick={initializeInArticleAdTools}
				icon={<MoveVertical/>}
				title="Move Ad Slots"
			/>}
			{changeInArticleVideoClickHandler && <DropdownItem
				onClick={changeInArticleVideoClickHandler}
				icon={<Video/>}
				title="Change In-Article Video"
			/>}
			{toggleConversationToolsClickHandler && <DropdownItem
				onClick={toggleConversationToolsClickHandler}
				icon={<Bubble/>}
				title="Toggle Conversation Tools"
			/>}
		</React.Fragment>
	</EnsureDefaultTheme>
);

export const PermalinkToolContainer = styled.div`
	display: flex;
	width: 100%;
	overflow-y: hidden;
	overflow-x: auto;
`;

export const PermalinkUserTools = styled.div`
	display: flex;
	position: relative;
`;

export const PermalinkShareTools = styled.div`
	height: 40px;
`;
