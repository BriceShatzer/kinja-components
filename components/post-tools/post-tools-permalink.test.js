import * as React from 'react';
import { shallow } from 'enzyme';

import { PostToolsPermalink } from './post-tools-permalink';

describe('<PostTools />', () => {
	it('should render move ad slots when callback is passed in', () => {
		const wrapper = shallow(
			<PostToolsPermalink
				initializeInArticleAdTools={() => {}}
			/>
		);
		expect(wrapper.dive()).toMatchSnapshot();
	});
});
