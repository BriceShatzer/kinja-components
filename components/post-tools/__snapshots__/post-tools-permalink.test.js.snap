// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTools /> should render move ad slots when callback is passed in 1`] = `
<ThemeProvider
  theme={[Function]}
>
  <DropdownItem
    icon={<MoveVerticalIcon />}
    onClick={[Function]}
    title="Move Ad Slots"
  />
</ThemeProvider>
`;
